### ISSUE TITLE
pygccjit test failures on 32bit architectures

### ISSUE THREAD

**Issue by doko42** [davidmalcolm/pygccjit#6]
seen with 0.4, at least with python 2.7:

test_get_type_error (test.ErrorTests) ... libgccjit.so: error: gcc_jit_context_get_type: unrecognized value for enum gcc_jit_types: -1
ok
test_new_block_error (test.ErrorTests) ... libgccjit.so: error: gcc_jit_function_new_block: cannot add block to an imported function
ok
test_new_function_error (test.ErrorTests) ... libgccjit.so: error: gcc_jit_context_new_function: name "contains a space" contains invalid character: ' '
ok
test_bf_aot (test.JitTests) ... ok
test_bf_jit (test.JitTests) ... ok
test_call_through_function_ptr (test.JitTests) ... ok
test_dereference (test.JitTests) ... ERROR
test_dump_reproducer (test.JitTests) ... ok
test_imported_function (test.JitTests) ... ok
test_opaque_struct (test.JitTests) ... ok
test_rvalue_from_ptr (test.JitTests) ... ERROR
test_set_logfile (test.JitTests) ... ok
test_square (test.JitTests) ... ok
test_sum_of_squares (test.JitTests) ... ok
test_union (test.JitTests) ... ok

======================================================================
ERROR: test_dereference (test.JitTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/«PKGBUILDDIR»/tests/test.py", line 114, in test_dereference
    nonnull_ptr = ctxt.new_rvalue_from_ptr(type_, id(self))
  File "/gccjit/gccjit.pyx", line 294, in gccjit._gccjit.Context.new_rvalue_from_ptr (gccjit/gccjit.c:5736)
OverflowError: Python int too large to convert to C long

======================================================================
ERROR: test_rvalue_from_ptr (test.JitTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/«PKGBUILDDIR»/tests/test.py", line 108, in test_rvalue_from_ptr
    nonnull_ptr = ctxt.new_rvalue_from_ptr(type_, id(self))
  File "/gccjit/gccjit.pyx", line 294, in gccjit._gccjit.Context.new_rvalue_from_ptr (gccjit/gccjit.c:5736)
OverflowError: Python int too large to convert to C long

----------------------------------------------------------------------
Ran 15 tests in 0.750s

FAILED (errors=2)

**Comment by davidmalcolm** [comment]
Does it work if you apply this patch?
```
diff --git a/gccjit/gccjit.pyx b/gccjit/gccjit.pyx
index 4e805de..7371a28 100644
--- a/gccjit/gccjit.pyx
+++ b/gccjit/gccjit.pyx
@@ -291,7 +291,7 @@ cdef class Context:
                                                              value)
         return RValue_from_c(self._c_ctxt, c_rvalue)
 
-    def new_rvalue_from_ptr(self, Type pointer_type, long value):
+    def new_rvalue_from_ptr(self, Type pointer_type, long long value):
         c_rvalue = c_api.gcc_jit_context_new_rvalue_from_ptr(self._c_ctxt,
                                                              pointer_type._get_c_type(),
                                                              <void *>value)
```

**Comment by doko42** [comment]
On 06/18/2015 02:15 AM, David Malcolm wrote:
> diff --git a/gccjit/gccjit.pyx b/gccjit/gccjit.pyx
> index 4e805de..7371a28 100644
> --- a/gccjit/gccjit.pyx
> +++ b/gccjit/gccjit.pyx
> @@ -291,7 +291,7 @@ cdef class Context:
>                                                               value)
>          return RValue_from_c(self._c_ctxt, c_rvalue)
>  
> -    def new_rvalue_from_ptr(self, Type pointer_type, long value):
> +    def new_rvalue_from_ptr(self, Type pointer_type, long long value):
>          c_rvalue = c_api.gcc_jit_context_new_rvalue_from_ptr(self._c_ctxt,
>                                                               pointer_type._get_c_type(),
>                                                               <void *>value)
> ```

yes, this fixes the tests.

**Comment by davidmalcolm** [comment]
Thanks; should be fixed in 541d3a6e1dbd9016d62c9069b2a4d1cad8fd9875
